### PR TITLE
Don't fail if tag has no parent version

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/giantswarm/issues/13410

### CODE REVIEW

**Review by MarcelMue — APPROVED** [review]
Works for me if I understand it correctly.

**Review by kopiczko — APPROVED** [review]

File: pkg/gen/input/workflows/internal/file/create_release.go [hunk 1]
```diff
@@ -217,8 +217,14 @@ jobs:
...
-          parent_version="$(git describe --tags --abbrev=0 HEAD^)"
+          parent_version="$(git describe --tags --abbrev=0 HEAD^ || true)"
           parent_version="${parent_version#v}" # Strip "v" prefix.
+
+          if [[ "$parent_version" = "" ]] ; then
+            echo "Unable to find a parent tag version. No branch to create."
+            exit 0
+          fi
+
           echo "current_version=$current_version parent_version=$parent_version"
```
Comment (kopiczko) [inline pkg/gen/input/workflows/internal/file/create_release.go:228]: We need changelog entry.

File: pkg/gen/input/workflows/internal/file/create_release.go [hunk 2]
```diff
@@ -217,8 +217,14 @@ jobs:
       - name: Create long-lived release branch
         run: |
           current_version="${{ needs.gather_facts.outputs.version }}"
-          parent_version="$(git describe --tags --abbrev=0 HEAD^)"
+          parent_version="$(git describe --tags --abbrev=0 HEAD^ || true)"
           parent_version="${parent_version#v}" # Strip "v" prefix.
+
+          if [[ "$parent_version" = "" ]] ; then
```
Comment (kopiczko) [inline pkg/gen/input/workflows/internal/file/create_release.go:223]: Feels like we should use `-z`.

File: pkg/gen/input/workflows/internal/file/create_release.go (hunk 1 above)
Comment (stone-z) [reply pkg/gen/input/workflows/internal/file/create_release.go:228]: Whoops - added

File: pkg/gen/input/workflows/internal/file/create_release.go (hunk 2 above)
Comment (stone-z) [reply pkg/gen/input/workflows/internal/file/create_release.go]: Yeah that's probably better. Changed this

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Generate language specific Makefiles in `gen makefile`.
 - Add required `--language` flag to `gen makefile`.
+- Do not try to create a previous release branch when tagging the first release.
```
Comment (kopiczko) [inline CHANGELOG.md]: This is  fixed

Comment (stone-z) [reply CHANGELOG.md]: Yeah, that got moved prior to merge

**Pull request merged**